docs/en/ingest-management/elastic-agent/uninstall-elastic-agent.asciidoc

```diff
@@ -22,17 +22,38 @@ stops and uninstalls any managed programs, such as {beats} and
 
 If you run into problems, see <<fleet-troubleshooting>>.
 
+If you are using DEB or RPM, you can use the package manager to remove the
+installed package.
+
 [discrete]
-== Uninstall on DEB or RPM
+== Remove {agent} files manually
+
+You might need to remove {agent} files manually if there's a failure during
+installation.
+
+To remove {agent} manually from your system:
+
+. <<unenroll-elastic-agent,Unenroll the agent>> if it's managed by {fleet}.
+
+. For standalone agents, back up any configuration files you want to preserve. 
 
-The `uninstall` command is not supported for DEB or RPM installations. To
-uninstall {agent} on DEB or RPM:
+. On your host, <<stop-elastic-agent,stop the agent>>. If any {agent}-related
+processes are still running, stop them too.
++
+TIP: Search for these processes and stop them if they're still running:
+`filebeat`, `metricbeat`, `fleet-server`, and `elastic-endpoint`.
 
-. If the agent is managed by {fleet}, <<unenroll-elastic-agent,unenroll the agent>>.
-. On your host, <<stop-elastic-agent,stop the agent>>.
-. Manually remove the agent files from your system.
+. Manually remove the {agent} files from your system. For example, if you're
+running {agent} on macOS, delete `/Library/Elastic/Agent/*`. Not sure where the
+files are installed? See <<installation-layout>>.
 
-//TODO: Confirm this procedure.
+. If you've configured the Elastic Endpoint integration, also remove the files
+installed for endpoint protection. The directory structure is similar to {agent},
+for example, `/Library/Elastic/Endpoint/*`.
++
+NOTE: When you remove the Elastic Endpoint integration from a macOS host, the
```

Comment:
@ferullo Can you confirm that this note is correct and necessary? I'm a little confused when I look at the endpoint docs because they [mention a kernel extension](https://www.elastic.co/guide/en/security/current/install-endpoint.html#enable-kernel-extension), not a system extension. Maybe it depends on the macOS version? Can you help me clarify this statement?

Reply:
@ferullo I'm going to comment this statement out in the source so I can get this merged today, but I'd really appreciate your input when you have a chance. Thanks!